ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/vm/MessageFrame.java

```diff
@@ -1026,6 +1045,7 @@ private void validate() {
       checkState(messageFrameStack != null, "Missing message frame message frame stack");
       checkState(worldState != null, "Missing message frame world state");
       checkState(initialGas != null, "Missing message frame initial getGasRemaining");
+      checkState(transactionHash != null, "Missing message frame transactionHash");
```

Comment:
There are a few cases where the MessageFrame.builder() is used that do not set the transactionHash, e.g. in AbstractCreateOperation.java line 105). These builders should fail because we are checking for null here. Does that mean that the code where these builders are used is dead code, or does it mean that we do not have tests that executes these builders, but they could be used and would fail?

Reply:
true - The main issue is in `VMReferenceTest` - because the transaction hash is not required for running the spec tests I have passed `Hash.EMPTY`.

The other option right now is to make the transaction hash optional - i.e. don't check if it is `null` - I don't like this personally as it is not consistent but there is no immediate negative impact.